docs/concepts/workloads/pods/disruptions.md

```diff
@@ -0,0 +1,262 @@
+Simultaneous failures of multiple nodes is less likely.  For even higher availability, use
+multi-zone clusters, spread applications across racks (using
+[anti-affinity](/docs/user-guide/node-selection/#inter-pod-affinity-and-anti-affinity-beta-feature))
+or across zones (if using a
+[multi-zone cluster](/docs/admin/multiple-zones).)
+
+The frequency of voluntary disruptions varies.  On a basic Kubernetes cluster, there are
+no voluntary disruptions at all.  However, your cluster admnistrator or hosting provider
+may run some additional services which cause voluntary disruptions.  For example,
+rolling out node software updates can cause voluntary updates.  Also, some implementations
+of cluster (node) autoscaling may cause voluntary disruptions to defragment and compact nodes.
+You cluster adminstrator or hosting provider should have documented what level of voluntary
+disruptions, if any, to expect.
+
+
+Kubernetes offers features to help run highly available applications at the same
+time as frequent voluntary disruptions.  We call this set of features
+*Disruption Budgets*.
+
+
+## How Disruption Budgets Work
+
+An Application Owner can create a `PodDisruptionBudget` object (PDB) for each application.
+A PDB limits the number pods of a replicated application that are down simultaneously due
+to voluntary disruptions.  For example, a quorum-based application would
+like to ensure that the number of replicas running is never brought below the
+number needed for a quorum, even temporarily. A web front end might want to
+ensure that the number of replicas serving load never falls below a certain
+percentage of the total, even briefly.  
+
+Cluster management tools can use the `Eviction API` to "safely delete" pods
+while respecting Pod Disruption Budgets.
+When a tool attempts to delete a pod using the Eviction API, Kuberetes checks if the proposed
+delete would leave the application with enough replicas.  If not, it rejects the request.
+The tool retries again later, and will typically succeed later.
+The `kubectl drain` tool uses the Eviction API.
+
+A PDB specifies the number of replicas that an application can tolerate having, relative to how
+many it is intended to have.  For example, a Deployment which has a `spec.replicas: 5` is
+supposed to have 5 pods at any given time.  If its PDB allows for there to be 4 at a time,
+then the Eviction API will allow voluntary disruption of one, but not two pods, at a time.
+
+The group of pods that comprise the application is specified using a label selector, the same
+as the one used by the application's controller (deployment, stateful-set, etc).
+
+The "intended" number of pods is computed from the `.spec.replicas` of the pods controller.
+The controller is discovered from the pods using the `.metadata.ownerReferences` of the object.
+
+PDBs cannot prevent [involuntary disruptions](#voluntary-and-involuntary-disruptions) from
```

Comment:
Is it worth noting that they can prevent an additional voluntary disruptions if involuntary ones have diminished capacity already?